src/fdc/MSXtar.cc

```diff
@@ -75,26 +83,28 @@ unsigned MSXtar::sectorToCluster(unsigned sector) const
 void MSXtar::parseBootSector(const MSXBootSector& boot)
 {
 	unsigned nbRootDirSectors = boot.dirEntries / DIR_ENTRIES_PER_SECTOR;
+	unsigned nrSectors = boot.nrSectors;
 	sectorsPerFat     = boot.sectorsFat;
 	sectorsPerCluster = boot.spCluster;
 
+	if (nrSectors == 0 && boot.extendedBootSignature == EBPB_SIGNATURE) {
+		nrSectors = boot.nrSectorsBig;
+	}
+
 	if (boot.bpSector != SECTOR_SIZE) {
 		throw MSXException("Illegal sector size: ", boot.bpSector);
 	}
-	if (boot.nrSectors == 0) { // TODO: check limits more accurately
-		throw MSXException("Illegal number of sectors: ", boot.nrSectors);
+	if (nrSectors == 0) { // TODO: check limits more accurately
+		throw MSXException("Illegal number of sectors: ", nrSectors);
 	}
 	if (boot.resvSectors == 0) {
 		throw MSXException("Illegal number of reserved sectors: ", boot.resvSectors);
 	}
-	if (boot.nrSides == 0) { // TODO: check limits more accurately
-		throw MSXException("Illegal number of sides: ", boot.nrSides);
-	}
```

Comment:
Was this intentional? Why is checking `nrSides` wrong or no longer needed?

Reply:
Because disk images formatted by Nextor _FDISK (at least the FAT16 ones) have this set to 0, I think for LBA type file systems.

Since it is not relevant to the correctness of the MSXtar implementation, there should be no need to check this one, otherwise there’d be a whole lot more complicated [CHS-related things](https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system#FATID) we could check.

Reply:
OK. This is useful information. Maybe add it as a comment?